tests/integration/test_unit.py

```diff
@@ -164,7 +164,7 @@ def check_results(results, out):
 async def test_scp():
     # ensure that asyncio.subprocess will work;
     try:
-        asyncio.get_child_watcher().attach_loop(jasyncio.get_running_loop())
+        asyncio.get_child_watcher().attach_loop(asyncio.get_running_loop())
```

Comment:
get_child_watcher is deprecated in Python 3.12.
Let's refactor this... somehow, I think it's safe to remove these 4 lines.
We're in control of how the integration test are run, I would imagine that pytest-asyncio uses the main thread by default, so the check if pointless.